controller/api/destination/traffic_split_adaptor.go

```diff
@@ -64,11 +66,19 @@ func (tsa *trafficSplitAdaptor) publish() {
 			overrides = append(overrides, dst)
 		}
 		merged.Spec.DstOverrides = overrides
-	}
-
-	if tsa.profile == nil && tsa.split == nil {
-		tsa.listener.Update(nil)
 	} else {
-		tsa.listener.Update(&merged)
+		// If there is no traffic split, always return a destination override
+		// so that it's known the host is a service.
+		overrides := []*sp.WeightedDst{}
+		id := strings.Split(tsa.id.String(), "/")
+		dst := &sp.WeightedDst{
+			Authority: fmt.Sprintf("%s.%s.svc.%s.:%d", id[1], tsa.id.Namespace, tsa.clusterDomain, tsa.port),
+			// Weights are expressed in decimillis: 10_000 represents 100%
+			Weight: resource.MustParse("10000m"),
```

Comment:
It doesn't really matter, since there's only one value, but this doesn't match the comment.  10_000m = 10 = 100_000 decimillis = 1000%

To set this to 100% you would want it to be "1000m" (or can you just do "1"?)